GNUmakefile

```diff
@@ -124,7 +124,12 @@ define yts_pass_fail
 endef
 export yts_pass_fail
 
-$(GOLANGCI-LINT): $(GO-DEPS)
+# Downloads and installs golangci-lint to LOCAL-BIN (.cache/local/bin).
+golangci-lint-install: $(GO-DEPS) $(YTS-DIR)
```

Comment:
I would prefer it like:
```
$(GOLANGCI-LINT): $(GO-DEPS) $(YTS-DIR)
        curl -sSfL $(GOLANGCI-LINT-INSTALL) | \
          bash -s -- -b $(LOCAL-BIN) $(GOLANGCI-LINT-VERSION)
        mv $(LOCAL-BIN)/golangci-lint $@

golangci-lint-install: $(GOLANGCI-LINT)
```
if that works.

Reply:
not really, because the Action is looking for a binary with name `golangci-lint`, when `$(GOLANGCI-LINT)` targets renames it to `golangci-lint-<version>`, that's why I split the target to two

Reply:
What do you mean by "the Action" in that sentence?

The `fmt` and `lint` Makefile rules depend on the value of `$(GOLANGCI-LINT)` (which contains a version, yes) and then they use that value directly via `$<`.

I don't get what needs `golangci-lint` (without the version in the name).

Reply:
I also wouldn't have the `$(GO-DEPS) $(YTS-DIR)` deps there as they appear to be unused in that rule.

Reply:
The `$(GO-DEPS)` and `$(YTS-DIR)` are deleted

Reply:
> I'm talking about `golangci-lint-action`, that is used in CI to run `golangci-lint`. The action expects `golangci-lint` binary in `PATH`. So, we need to be able to install it without renaming. Nothing should be changed for developers, all the targets must use `$(GOLANGCI-LINT)`, and only the CI will use `golangci-lint-install` target to install it as it is.

I see. Thanks for explaining. I'd still do it a little different. I'll push a commit back to you.